pymongo/network_layer.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2015-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Internal network layer helper methods."""
+from __future__ import annotations
+
+import asyncio
+import socket
+import struct
+from typing import (
+    TYPE_CHECKING,
+    Union,
+)
+
+from pymongo import ssl_support
+
+if TYPE_CHECKING:
+    from pymongo.pyopenssl_context import _sslConn
```

Comment:
That fails the typechecker: 

```
pymongo/asynchronous/pool.py:1034: error: Argument 1 to "async_sendall" has incompatible type "Union[socket, _sslConn]"; expected
"Union[socket, SSLSocket]"  [arg-type]
                await async_sendall(self.conn, message)
```

Reply:
Interesting, maybe this is because we always install pyopenssl when type checking. I'll open a new ticket since I just realized this is an existing problem in the types.